packages/firestore/exp/src/api/database.ts

```diff
@@ -80,16 +90,71 @@ export class Firestore extends LiteFirestore
       );
 
       this._firestoreClientPromise = firestoreClient
-        .start(this._componentProvider, { durable: false })
+        .start(this._componentProvider, this._persistenceSettings)
         .then(() => firestoreClient);
     }
 
     return this._firestoreClientPromise;
   }
 
+  // TODO(firestorexp): Factor out MultiTabComponentProvider and remove
+  // `synchronizeTabs` argument
+  _enablePersistence(
+    persistenceProvider: ComponentProvider,
+    synchronizeTabs: boolean
+  ): Promise<void> {
+    if (this._firestoreClientPromise) {
+      throw new FirestoreError(
+        Code.FAILED_PRECONDITION,
+        'Firestore has already been started and persistence can no longer ' +
+          'be enabled. You can only call enable persistence before calling ' +
+          'any other methods on a Firestore object.'
+      );
+    }
+
+    const settings = this._getSettings();
+    this._persistenceSettings = {
+      durable: true,
+      synchronizeTabs,
+      forceOwningTab: false,
+      cacheSizeBytes:
+        settings.cacheSizeBytes ?? LruParams.DEFAULT_CACHE_SIZE_BYTES
+    };
+    this._componentProvider = persistenceProvider;
+
+    // TODO(firestorexp): Add support for Persistence fallback
+    return this._getFirestoreClient().then(() => {});
+  }
+
   delete(): Promise<void> {
     return terminate(this);
   }
+
+  _clearPersistence(): Promise<void> {
+    if (this._firestoreClientPromise !== undefined && !this._terminated) {
+      throw new FirestoreError(
+        Code.FAILED_PRECONDITION,
+        'Persistence cannot be cleared after this Firestore instance is initialized.'
```

Comment:
nit: This statement is not entirely correct. Consider this rewording:

*Persistence can only be cleared before the Firestore instance is initialized or after it is terminated.*

Reply:
Udpated